specs/core/0_beacon-chain.md

```diff
@@ -71,122 +70,134 @@ The primary source of load on the beacon chain are "attestations". Attestations
+* At most `1/MAX_CHURN_QUOTIENT` of the validators can change during each validator registry change.
+* The `MAX_WITHDRAWALS_PER_EPOCH` parameter correspond to 5.2m ETH in ~6 months.
 
-## PoW chain registration contract
+## Ethereum 1.0 chain deposit contract
 
-The initial deployment phases of Ethereum 2.0 are implemented without consensus changes to the PoW chain. A registration contract is added to the PoW chain to deposit ETH. This contract has a `registration` function which takes as arguments `pubkey`, `withdrawal_credentials`, `randao_commitment` as defined in a `ValidatorRecord` below. A BLS `proof_of_possession` of types `bytes` is given as a final argument.
+The initial deployment phases of Ethereum 2.0 are implemented without consensus changes to the Ethereum 1.0 chain. A deposit contract is added to the Ethereum 1.0 chain to deposit ETH. This contract has a `deposit` function which takes as arguments `pubkey`, `withdrawal_credentials`, `randao_commitment` as defined in a `ValidatorRecord` below. A BLS `proof_of_possession` of types `bytes` is given as a final argument.
 
-The registration contract emits a log with the various arguments for consumption by the beacon chain. It does not do validation, pushing the registration logic to the beacon chain. In particular, the proof of possession (based on the BLS12-381 curve) is not verified by the registration contract.
+The deposit contract emits a log with the various arguments for consumption by the beacon chain. It does little validation, pushing the deposit logic to the beacon chain. In particular, the proof of possession (based on the BLS12-381 curve) is not verified by the deposit contract.
 
 ### Contract code in Vyper
 
-The beacon chain is initialized when a condition is met inside a contract on the existing PoW chain. This contract's code in Vyper is as follows:
+The beacon chain is initialized when a condition is met inside a contract on the existing Ethereum 1.0 chain. This contract's code in Vyper is as follows:
 
 ```python
-DEPOSITS_FOR_CHAIN_START: constant(uint256) = 2**14
-DEPOSIT_SIZE: constant(uint256) = 32  # ETH
-MIN_TOPUP_SIZE: constant(uint256) = 1  # ETH
+MAX_DEPOSITS_FOR_CHAIN_START: constant(uint256) = 2**14
+MAX_DEPOSIT: constant(uint256) = 32  # ETH
+MIN_DEPOSIT: constant(uint256) = 1  # ETH
 GWEI_PER_ETH: constant(uint256) = 10**9
 POW_CONTRACT_MERKLE_TREE_DEPTH: constant(uint256) = 32
 SECONDS_PER_DAY: constant(uint256) = 86400
 
-HashChainValue: event({previous_receipt_root: bytes32, data: bytes[2064], total_deposit_count: uint256})
+HashChainValue: event({previous_receipt_root: bytes32, data: bytes[2064], max_deposit_count: uint256})
 ChainStart: event({receipt_root: bytes32, time: bytes[8]})
 
 receipt_tree: bytes32[uint256]
-total_deposit_count: uint256
+max_deposit_count: uint256
```

Comment:
IMO naming something `max_` makes it looks like a constant.

Reply:
Addressed :)